File: android/src/androidTest/java/org/ligi/gobandroid_hd/uitest/TheGoActivity.java
package org.ligi.gobandroid_hd.uitest;

import android.test.suitebuilder.annotation.MediumTest;

import com.squareup.spoon.Spoon;

import org.ligi.gobandroid_hd.R;
import org.ligi.gobandroid_hd.ui.GoActivity;
import org.ligi.gobandroid_hd.base.BaseIntegration;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;

public class TheGoActivity extends BaseIntegration<GoActivity> {

    public TheGoActivity() {
        super(GoActivity.class);
    }

    @MediumTest
    public void testThatGoBoardIsThere() {
        final GoActivity activity = getActivity();

        Spoon.screenshot(activity, "go_activity");
        onView(withId(R.id.go_board)).check(matches(isDisplayed()));
    }


}

File: android/src/androidTest/java/org/ligi/gobandroid_hd/uitest/TheGoActivity.kt
package org.ligi.gobandroid_hd.uitest

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.runner.AndroidJUnit4
import com.jraska.falcon.FalconSpoon
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.ui.GoActivity
import org.ligi.trulesk.TruleskActivityRule

@RunWith(AndroidJUnit4::class)
class TheGoActivity {

    @get:Rule
    val rule = TruleskActivityRule(GoActivity::class.java)

    @Test
    fun testThatGoBoardIsThere() {
        FalconSpoon.screenshot(rule.activity, "go_activity")
        onView(withId(R.id.go_board)).check(matches(isDisplayed()))
    }

}
